src/model/HVACComponent_Impl.hpp

```diff
@@ -114,6 +118,11 @@ namespace model {
 
       virtual std::vector<std::string> emsInternalVariableNames() const override;
 
+      virtual ComponentType componentType() const = 0;
+      virtual std::vector<FuelType> coolingFuelTypes() const = 0;
+      virtual std::vector<FuelType> heatingFuelTypes() const = 0;
+      virtual std::vector<AppGFuelType> appGHeatingFuelTypes() const = 0;
```

Comment:
As you can see in the implementation, I'm often using a `std::set` internally so that the returned `std::vector` is actually unique (and sorted). 
I didn't go with `std::set` because I was unsure how SWIG would play with a `std::set`...

Would you prefer `std::set? @kbenne @tijcolem @asparke2 please?

Reply:
I don't know enough C++ to comment on the implementation sidee, but on the return type you get from the API I'd prefer a std::vector<Whatever>.